### ISSUE TITLE
answer for 5.7 - Exercise 7

### ISSUE THREAD

**Issue by dongzhuoer** [jrnold/r4ds-exercise-solutions#41]
we need to "Find all destinations that are flown by at least two carriers"

```r
(two_more_carriers <- flights %>%
    group_by(dest) %>% 
    mutate(n_carrier = n_distinct(carrier)) %>% 
    filter(n_carrier >= 2) %>% 
    select(-n_carrier) %>%
    ungroup())
```

```
# A tibble: 325,397 x 19
    year month   day dep_time sched_dep_time dep_delay arr_time sched_arr_time arr_delay carrier flight tailnum origin dest  air_time
   <int> <int> <int>    <int>          <int>     <dbl>    <int>          <int>     <dbl> <chr>    <int> <chr>   <chr>  <chr>    <dbl>
 1  2013     1     1      517            515         2      830            819        11 UA        1545 N14228  EWR    IAH        227
 2  2013     1     1      533            529         4      850            830        20 UA        1714 N24211  LGA    IAH        227
 3  2013     1     1      542            540         2      923            850        33 AA        1141 N619AA  JFK    MIA        160
 4  2013     1     1      544            545        -1     1004           1022       -18 B6         725 N804JB  JFK    BQN        183
 5  2013     1     1      554            600        -6      812            837       -25 DL         461 N668DN  LGA    ATL        116
 6  2013     1     1      554            558        -4      740            728        12 UA        1696 N39463  EWR    ORD        150
 7  2013     1     1      555            600        -5      913            854        19 B6         507 N516JB  EWR    FLL        158
 8  2013     1     1      557            600        -3      709            723       -14 EV        5708 N829AS  LGA    IAD         53
 9  2013     1     1      557            600        -3      838            846        -8 B6          79 N593JB  JFK    MCO        140
10  2013     1     1      558            600        -2      753            745         8 AA         301 N3ALAA  LGA    ORD        138
# ... with 325,387 more rows, and 4 more variables: distance <dbl>, hour <dbl>, minute <dbl>, time_hour <dttm>
```

then "rank the carriers"

```r
two_more_carriers %>% 
    group_by(carrier) %>%
    summarise(n_dest = n_distinct(dest)) %>%
    arrange(desc(n_dest))
```

```
# A tibble: 16 x 2
   carrier n_dest
   <chr>    <int>
 1 EV          51
 2 9E          48
 3 UA          42
 4 DL          39
 5 B6          35
 6 AA          19
 7 MQ          19
 8 WN          10
 9 OO           5
10 US           5
11 VX           4
12 YV           3
13 FL           2
14 AS           1
15 F9           1
16 HA           1
```

Actually it took me a while to understand your code. I found that `count()` can be quite confused when you have to group by two variables. Luckily, `n_distinct()` and customize the new variable name by my own `summarise()` makes the process quite clear and straight-forward.

**Comment by jrnold** [comment]
I'll add that alternative code. But I don't understand how `count()` is not pretty clear, regardless of the number of variables being grouped by.

**Comment by jrnold** [comment]
This code is clearer and closer to the type of code presented in the book. I'll add a new solution similar to this code. It would be even simpler with `distinct()`, but I don't  see it having been introduced in R4DS up to that point.

**Comment by jrnold** [comment]
Fixed in 13e0ffdd71eb3aeea2f6cb018e8c5afc3302d544
